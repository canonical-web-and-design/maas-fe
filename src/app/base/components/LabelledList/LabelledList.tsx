import type { ReactNode } from "react";

import { List } from "@canonical/react-components";
import classNames from "classnames";

type LabelledListItem = {
  label: ReactNode;
  value: ReactNode;
};

type Props = {
  className?: string;
  items: LabelledListItem[];
};

const LabelledList = ({ className, items }: Props): JSX.Element => {
  return (
    <List
      className={classNames("p-list--labelled", className)}
      items={items.map(({ label, value }) => (
        <>
          <div className="p-list__item-label">{label}</div>
          <div className="p-list__item-value">{value}</div>
        </>
      ))}
    />
  );
};

export default LabelledList;
