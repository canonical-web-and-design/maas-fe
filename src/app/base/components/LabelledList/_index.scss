@mixin LabelledList {
  .p-list--labelled {
    @extend %vf-list;

    .p-list__item {
      @extend %vf-list-item;
      display: flex;
      // Vanilla lists have a different padding inside forms, but since this is
      // a custom component we need to make sure it doesn't change.
      padding: $spv--small 0 !important;
      word-wrap: break-word;
    }

    .p-list__item-label {
      color: $color-mid-dark;
      flex: 1 0 0;
      margin-right: $sph--small;
      text-align: right;
    }

    .p-list__item-value {
      flex: 2 1 0;
      word-break: break-word;
    }
  }
}
