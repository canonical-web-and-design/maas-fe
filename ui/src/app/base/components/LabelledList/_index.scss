@mixin LabelledList {
  .p-list--labelled {
    @extend %vf-list;

    .p-list__item {
      @extend %vf-list-item;
      display: flex;
      // Vanilla lists have a different padding inside forms, but since this is
      // a custom component we need to make sure it doesn't change.
      padding: $spv-inner--x-small--scaleable 0 !important;
      word-wrap: break-word;
    }

    .p-list__item-label {
      color: $color-mid-dark;
      flex: 1 0 0;
      margin-right: $sph-outer;
      text-align: right;
    }

    .p-list__item-value {
      flex: 2 1 0;
    }
  }
}
