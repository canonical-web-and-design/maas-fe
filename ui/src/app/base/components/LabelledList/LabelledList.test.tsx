import { shallow } from "enzyme";
import React from "react";

import LabelledList from "./LabelledList";

describe("LabelledList ", () => {
  it("renders", () => {
    const wrapper = shallow(
      <LabelledList items={[{ label: "Item", value: "Value" }]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("can add additional classes", () => {
    const wrapper = shallow(
      <LabelledList className="extra-class" items={[]} />
    );
    const className = wrapper.prop("className");
    expect(className.includes("p-list--labelled")).toBe(true);
    expect(className.includes("extra-class")).toBe(true);
  });
});
