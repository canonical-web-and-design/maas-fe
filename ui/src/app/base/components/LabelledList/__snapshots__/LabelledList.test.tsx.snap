// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledList  renders 1`] = `
<List
  className="p-list--labelled"
  items={
    Array [
      <React.Fragment>
        <div
          className="p-list__item-label"
        >
          Item
        </div>
        <div
          className="p-list__item-value"
        >
          Value
        </div>
      </React.Fragment>,
    ]
  }
/>
`;
